### ISSUE TITLE
xpath selector doesn't work if search context is an element (not a driver)

### ISSUE THREAD

**Issue by vgrigoruk** [appium/appium#2558]
driver.find_element(:xpath, '//UIATableView[1]')

Appium log:

```
debug: Appium request initiated at /wd/hub/session/43dc50b4-a81b-4ec0-9f00-add2ea9b4e38/element
debug: Request received with params: {"using":"xpath","value":"//UIATableView[1]"}
info: Pushing command to appium work queue: "au.mainApp().getTreeForXML()"
debug: Sending command to instruments: au.mainApp().getTreeForXML()
info: [INSTSERVER] Sending command to instruments: au.mainApp().getTreeForXML()
info: [INSTSERVER] Socket data received (8186 bytes)
info: [INSTSERVER] Socket data received (8192 bytes)
info: [INSTSERVER] Socket data received (5420 bytes)
info: [INSTSERVER] Socket data being routed for 'cmd' event
info: [INSTSERVER] Got result from instruments: {"status":0,"value":"{\"UIAApplication\":{\"@\":{\"name\":\"Podio\",\"label\":\"Podio\",\"value\":null,\"dom\":null,\"enabled\":true,\"valid\":true,\"visible\":true,\"hint\":null,\"path\":\"/0\",\"x\":0,\"y\":20,\"width\":320,\"height\":548},\">\":[{\"UIAWindow\":{\"@\":{\"name\":null,\"label\":null
info: Pushing command to appium work queue: "au.getElementByIndexPath('/0/0/3')"
debug: Sending command to instruments: au.getElementByIndexPath('/0/0/3')
info: [INSTSERVER] Sending command to instruments: au.getElementByIndexPath('/0/0/3')
info: [INSTSERVER] Socket data received (62 bytes)
info: [INSTSERVER] Socket data being routed for 'cmd' event
info: [INSTSERVER] Got result from instruments: {"status":0,"value":{"ELEMENT":"48"}}
info: Responding to client with success: {"status":0,"value":{"ELEMENT":"48"},"sessionId":"43dc50b4-a81b-4ec0-9f00-add2ea9b4e38"}
POST /wd/hub/session/43dc50b4-a81b-4ec0-9f00-add2ea9b4e38/element 200 619ms - 110b
```

Then trying to do search for table cell inside a UIATableView using chained queries.

    driver.find_element(:xpath, '//UIATableView[1]').find_element(:xpath, '//UIATableCell[1]/UIAElement[1]').text

And the log is:
```
debug: Appium request initiated at /wd/hub/session/43dc50b4-a81b-4ec0-9f00-add2ea9b4e38/element
debug: Request received with params: {"using":"xpath","value":"//UIATableView[1]"}
info: Pushing command to appium work queue: "au.mainApp().getTreeForXML()"
debug: Sending command to instruments: au.mainApp().getTreeForXML()
info: [INSTSERVER] Sending command to instruments: au.mainApp().getTreeForXML()
info: [INSTSERVER] Socket data received (8186 bytes)
info: [INSTSERVER] Socket data received (8192 bytes)
info: [INSTSERVER] Socket data received (5420 bytes)
info: [INSTSERVER] Socket data being routed for 'cmd' event
info: [INSTSERVER] Got result from instruments: {"status":0,"value":"{\"UIAApplication\":{\"@\":{\"name\":\"Podio\",\"label\":\"Podio\",\"value\":null,\"dom\":null,\"enabled\":true,\"valid\":true,\"visible\":true,\"hint\":null,\"path\":\"/0\",\"x\":0,\"y\":20,\"width\":320,\"height\":548},\">\":[{\"UIAWindow\":{\"@\":{\"name\":null,\"label\":null
info: Pushing command to appium work queue: "au.getElementByIndexPath('/0/0/3')"
debug: Sending command to instruments: au.getElementByIndexPath('/0/0/3')
info: [INSTSERVER] Sending command to instruments: au.getElementByIndexPath('/0/0/3')
info: [INSTSERVER] Socket data received (62 bytes)
info: [INSTSERVER] Socket data being routed for 'cmd' event
info: [INSTSERVER] Got result from instruments: {"status":0,"value":{"ELEMENT":"49"}}
info: Responding to client with success: {"status":0,"value":{"ELEMENT":"49"},"sessionId":"43dc50b4-a81b-4ec0-9f00-add2ea9b4e38"}
POST /wd/hub/session/43dc50b4-a81b-4ec0-9f00-add2ea9b4e38/element 200 561ms - 110b
debug: Appium request initiated at /wd/hub/session/43dc50b4-a81b-4ec0-9f00-add2ea9b4e38/element/49/element
debug: Request received with params: {"using":"xpath","value":"//UIATableCell[1]/UIAElement[1]"}
info: Pushing command to appium work queue: "au.getElement('49').getTreeForXML()"
debug: Sending command to instruments: au.getElement('49').getTreeForXML()
info: [INSTSERVER] Sending command to instruments: au.getElement('49').getTreeForXML()
info: [INSTSERVER] Socket data received (2605 bytes)
info: [INSTSERVER] Socket data being routed for 'cmd' event
info: [INSTSERVER] Got result from instruments: {"status":0,"value":"{\"UIATableView\":{\"@\":{\"name\":null,\"label\":null,\"value\":\"rows 1 to 3 of 3\",\"dom\":null,\"enabled\":true,\"valid\":true,\"visible\":true,\"hint\":null,\"path\":\"/0\",\"x\":0,\"y\":64,\"width\":320,\"height\":504},\">\":[{\"UIATableCell\":{\"@\":{\"name\":\"All Field 
info: Pushing command to appium work queue: "au.getElementByIndexPath('/0/0/1')"
debug: Sending command to instruments: au.getElementByIndexPath('/0/0/1')
info: [INSTSERVER] Sending command to instruments: au.getElementByIndexPath('/0/0/1')
info: [INSTSERVER] Socket data received (62 bytes)
info: [INSTSERVER] Socket data being routed for 'cmd' event
info: [INSTSERVER] Got result from instruments: {"status":0,"value":{"ELEMENT":"50"}}
info: Responding to client with success: {"status":0,"value":{"ELEMENT":"50"},"sessionId":"43dc50b4-a81b-4ec0-9f00-add2ea9b4e38"}
POST /wd/hub/session/43dc50b4-a81b-4ec0-9f00-add2ea9b4e38/element/49/element 200 282ms - 110b
debug: Appium request initiated at /wd/hub/session/43dc50b4-a81b-4ec0-9f00-add2ea9b4e38/element/50/text
debug: Request received with params: {}
info: Pushing command to appium work queue: "au.getElement('50').text()"
debug: Sending command to instruments: au.getElement('50').text()
info: [INSTSERVER] Sending command to instruments: au.getElement('50').text()
info: [INSTSERVER] Socket data received (102 bytes)
info: [INSTSERVER] Socket data being routed for 'cmd' event
info: [INSTSERVER] Got result from instruments: {"status":0,"value":"UH OH! LOADING FAILED, CHECK YOUR INTERNET CONNECTION."}
info: Responding to client with success: {"status":0,"value":"UH OH! LOADING FAILED, CHECK YOUR INTERNET CONNECTION.","sessionId":"43dc50b4-a81b-4ec0-9f00-add2ea9b4e38"}
GET /wd/hub/session/43dc50b4-a81b-4ec0-9f00-add2ea9b4e38/element/50/text 200 118ms - 141b
```

It seems like this query is not built correctly:
    
    au.getElementByIndexPath('/0/0/1')

Instead, it should be:

    au.getElementByIndexPath('/0/0/3/0/1') 

Note: running from sources d9f0ae7d718e06959854b9b1d3a23306adbee212

**Comment by chrigrahcisco** [comment]
Appium.app 1.3.6
iOS8.2 iPhone6 simulator
Python bindings

I believe I just discovered this same thing today when attempting to work with UIATableCell within a dynamic UIATableView by attempting to use `find_element_by_xpath()` against the cell. 

    table = self.driver.find_element_by_xpath(table_xpath)
    cells = table.find_elements_by_class_name('UIATableCell')
    for i in range(len(cells)):
        cells[i].find_element_by_xpath("UIAButton").click()

The click would fail with NoSuchElementException

However, find_element_by_class_name works just fine in the same scenario and I'm using that plus some element indexes as a workaround.

**Comment by vgrigoruk** [comment]
@chrigrahcisco, "UIAButton" is not a valid xpath. ".//UIAButton" is a valid one.

**Comment by chrigrahcisco** [comment]
Excellent point. I probably hand typed that instead of copy-paste from actual code. At this point I no longer have that code and I have no idea what I was using but hopefully not just "UIAButton". :)

**Comment by artkoshelev** [comment]
any chances to see this feature soon?

**Comment by LoveleenSaini** [comment]
Any update on this feature?

**Comment by guysmoilov** [comment]
I've created a pull request for this feature on android, so far I'm being ignored:
https://github.com/appium/appium-android-driver/pull/111
https://github.com/appium/appium-android-bootstrap/pull/17

**Comment by imurchie** [comment]
I apologize for the delay in getting to those PRs. Both are merged and will be in the next release (or, if you're running from source, a little later today, as I'm waiting for one more commit to get in before publishing `appium-android-driver`. I'll post here when running from source should work.

**Comment by imurchie** [comment]
The change should be available in `appium-android-driver@1.7.3`.

**Comment by SrinivasanTarget** [comment]
@triager please close

**Comment by Moustakas42** [comment]
Is something like 

cells[i].find_element_by_xpath(".//UIAButton") 

supposed to work now?

**Comment by qual** [comment]
I tried with ./* but its not working. It is supposed to return the direct children of the element but it returns 0 elements. Can someone please check why ./* doesn't work ?

**Comment by mykola-mokhnach** [comment]
@qual This should work properly in Appium 1.6.4-beta

**Comment by qual** [comment]
OK. Thanks. I am using Appium 1.6.0. I think thats why it is not working. Is this implemented in Appium-android-driver ? Can we install latest version of Appium-android-driver with Appium 1.6.0.
As you mentioned, this is fixed in 1.6.4-beta. Does this mean that it has latest Appium-android-driver built-in ?
